scripts/globals/crafting.lua

```diff
@@ -195,56 +189,113 @@ xi.crafting.unionRepresentativeEventUpdateRenounce = function(player, craftID)
 end
 
 xi.crafting.unionRepresentativeTriggerRenounceCheck = function(player, eventId, realSkill, rankCap, param3)
-    if player:getLocalVar("renounceDialog") == 0 then
+    if player:getLocalVar("skipRenounceDialog") == 0 then
         local count   = 0
         local bitmask = 0
 
         for craftID = xi.skill.WOODWORKING, xi.skill.COOKING do
             local rank = player:getSkillRank(craftID)
-            if rank < 7 then
-                bitmask = bit.bor(bitmask, bit.lshift(1, craftID - 48))
-            else
+
+            if rank >= xi.crafting.rank.ARTISAN then
                 count = count + 1
+            else
+                -- This needs checking. Probably made-up.
+                bitmask = bit.bor(bitmask, bit.lshift(1, craftID - 48))
             end
         end
 
-        if count > 1 then
-            player:setLocalVar("renounceDialog", 1)
+        if count >= 2 then
+            player:setLocalVar("skipRenounceDialog", 1)
+
+            -- TODO: Param 3 is taken directly from captures, but now we know this is incorrect.
+            -- TODO: bitmask needs checking.
             player:startEvent(eventId, VanadielTime(), realSkill, rankCap, param3, 0, 0, count, bitmask)
+
             return true
         end
     end
 
     return false
 end
 
+--------------------------------------------------
+-- Guild Point NPCs (Union Representatives)
+--------------------------------------------------
+xi.crafting.unionRepresentativeTrade = function(player, npc, trade, csid, guildID)
+    local _, remainingPoints = player:getCurrentGPItem(guildID)
+    local ID                 = zones[player:getZoneID()]
+
+    if player:getCharVar('[GUILD]currentGuild') - 1 == guildID then
+        if remainingPoints == 0 then
+            player:messageText(npc, ID.text.NO_MORE_GP_ELIGIBLE)
+        else
+            local totalPoints = 0
+            for i = 0, 8 do
+                local items, points = player:addGuildPoints(guildID, i)
+
+                if items ~= 0 and points ~= 0 then
+                    totalPoints = totalPoints + points
+                    trade:confirmSlot(i, items)
+                end
+            end
+
+            if totalPoints > 0 then
+                player:confirmTrade()
+                player:startEvent(csid, totalPoints)
+            end
+        end
+    end
+end
+
+xi.crafting.unionRepresentativeTrigger = function(player, guildId, csid, currency, keyitems)
+    local gpItem, remainingPoints = player:getCurrentGPItem(guildId)
+    local rank                    = player:getSkillRank(guildId + 48)
+    local cap                     = (rank + 1) * 10
+    local kibits                  = 0
```

Comment:
Needs some more sane variable naming for keyItem bits, and the k,i iteration below.